lib/node_modules/@stdlib/math/base/special/ellipj/test/fixtures/cpp/runner.cpp

```diff
@@ -0,0 +1,371 @@
+*
+* @param f     file to write to
+* @param name  array name
+* @param x     data
+* @param len   array length
+*/
+void write_named_array_i32( FILE *f, const char *name, const int *x, const unsigned int len ) {
+	fprintf( f, "\"%s\":[", name );
+	write_array_i32( f, x, len );
+	fprintf( f, "]" );
+}
+
+/**
+* Writes data to a file as JSON.
+*
+* ## Notes
+*
+* -   This function SHOULD be tailored to the input data (e.g., input types, output types, number of arguments, etc) and may vary from use case to use case.
+*
+*
+* @param f                    file to write to
+* @param x                    first parameter
+* @param y                    complete elliptic integral of the first kind
+* @param len                  array length
+*/
+void write_data_as_json( FILE *f, const double *u, const double *m, const double *sn, const double *cn, const double *dn, const unsigned int len ) {
+	fprintf( f, "{" );
+	write_named_array_f64( f, "u", u, len );
+	fprintf( f, "," );
+	write_named_array_f64( f, "m", m, len );
+	fprintf( f, "," );
+	write_named_array_f64( f, "sn_expected", sn, len );
+	fprintf( f, "," );
+	write_named_array_f64( f, "cn_expected", cn, len );
+	fprintf( f, "," );
+	write_named_array_f64( f, "dn_expected", dn, len );
+	fprintf( f, "}" );
+}
+
+/**
+* Generates test fixtures.
+*
+* @param u     first parameter
+* @param m     modulus
+* @param len   number of values in the domain
+* @param name  output filename
+*/
+void generate( double *u, double *m, const unsigned int len, const char *name ) {
+	double *sn;
+	double *cn;
+	double *dn;
+	double k;
+	unsigned int i;
+	FILE *f;
+
+	// Allocate an output array:
+	sn = (double*) malloc( len * sizeof(double) );
+	cn = (double*) malloc( len * sizeof(double) );
+	dn = (double*) malloc( len * sizeof(double) );
+	if ( sn == NULL ) {
+		printf( "Error allocating memory.\n" );
+		exit( 1 );
+	}
+
+	// Generate fixture data:
+	for ( i = 0; i < len; i++ ) {
+		if ( m[ i ] >= 0.0 ) {
+			// Boost uses a different convention for the argument so that we must use a square root:
+			k = sqrt( m[ i ] );
+			sn[ i ] = jacobi_sn( k, u[ i ] );
+			cn[ i ] = jacobi_cn( k, u[ i ] );
+			dn[ i ] = jacobi_dn( k, u[ i ] );
+		} else {
+			/*
+			* Boost uses k as the argument, which is the square root of m. That means negative values of m require computation in terms of imaginary values of k. Instead, Equation 17.4.17 from Abramowitz and Stegun give the transformation
+			*
+			*   F(phi|-m) = (1+m)^(-1/2) K(m/(1+m)) - (1+m)^(-1/2) F(pi/2-phi|m/(1+m))
+			*
+			* Since K(m) is equivalent to F(phi|m), the above reduces to
+			*
+			*   F(phi|-m) = (1+m)^(-1/2) K(m/(1+m))
+			*
+			* with which we can then compute values with a negative argument in terms of values with a positive argument.
+			*/
+			/*m = abs( m );
+			k = sqrt( m/(1.0+m) );
+			sn[ i ] = jacobi_sn( k ) / sqrt( 1.0+m );
+			cn[ i ] = jacobi_cn( k ) / sqrt( 1.0+m );
+			dn[ i ] = jacobi_dn( k ) / sqrt( 1.0+m );
```

Comment:
Is leaving this commented intentional?

Reply:
Ah, thank you! I think I meant to just delete that. The transformation is straightforward, so I thought maybe it would make more sense to rigorously test values in the range `0 ≤ m ≤ 1` and the use wolfram spot checks for `m < 0` and `m > 1`.

Reply:
Fine with me. Whatever you think best.